src/controller.ts

```diff
@@ -174,7 +185,7 @@ export class Controller {
       return;
     }
 
-    const tree = parseSource(contents);
+    const tree = parseSource(contents, folder, uri, this.testSpecifiers);
```

Comment:
Kind of a gross API, but I wanted to keep the code structure as similar as possible without getting your feedback.  Making these parameters optional meant most of the tests didn't need change.